composeApp/src/androidMain/kotlin/composables/FileChooser.android.kt

```diff
@@ -37,20 +38,25 @@ class FileChooserLifecycleObserver(private val registry: ActivityResultRegistry,
 
     private fun writeFile(contentResolver: ContentResolver, uri: Uri?) {
         try {
-            if (uri != null) {
-                contentResolver.openFileDescriptor(uri, "w")?.use { fileDescriptor ->
-                    FileOutputStream(fileDescriptor.fileDescriptor).use {
-                        it.write(
-                            ("Overwritten at ${System.currentTimeMillis()}\n")
-                                .toByteArray()
-                        )
-                    }
-                }
+            val rom : Any = saveROM()
+
+            if (rom is ByteArray) {
```

Comment:
Is the idea here to have an `if (rom is NDSRom)` in the future? I'm wondering if `saveROM()` could pass a function back instead of the rom.

Reply:
Can you build around the idea of a "sealed class" ?
```
sealed class Rom {
    data class ByteArrayRom(val bytes: ByteArray) : Rom()
    data class NDSRom(val ndsRom: NDSRom) : Rom()
    // Add more ROM types if needed
}

val rom: Rom = saveROM()

when (rom) {
    is Rom.ByteArrayRom -> saveBytes(contentResolver, uri, rom.bytes)
    is Rom.NDSRom -> saveNDSRom(contentResolver, uri, rom.ndsRom)
}
```